[galactic] spatial-launch.py fails when not providing Magnetometer corrections

## Bug Report
After the switch to ROS2 Galactic, the `spatial-launch.py` (and probably also the magnetometer) fails when not providing magnetometer corrections.

### Reproduction
Simply calling `ros2 launch phidgets_spatial spatial-launch.py` yields:
```
[INFO] [component_container-1]: process started with pid [4695]
[component_container-1] 1626088543.095012 [0] component_: using network interface eth0 (udp/192.168.1.210) selected arbitrarily from: eth0, docker0
[component_container-1] [INFO] [1626088543.296766218] [phidget_container]: Load Library: /home/angsa/ros_ws/install/phidgets_spatial/lib/libphidgets_spatial.so
[component_container-1] [INFO] [1626088543.309701892] [phidget_container]: Found class: rclcpp_components::NodeFactoryTemplate<phidgets::SpatialRosI>
[component_container-1] [INFO] [1626088543.310028374] [phidget_container]: Instantiate class: rclcpp_components::NodeFactoryTemplate<phidgets::SpatialRosI>
[component_container-1] [INFO] [1626088543.336890208] [phidgets_spatial]: Starting Phidgets Spatial
[component_container-1] [ERROR] [1626088543.536905484] [phidget_container]: Component constructor threw an exception: expected [double] got [not set]
[ERROR] [launch_ros.actions.load_composable_nodes]: Failed to load node 'phidgets_spatial' of type 'phidgets::SpatialRosI' in container '/phidget_container': Component constructor threw an exception: expected [double] got [not set]
```

### Reason
The declare_parameter here fails:
https://github.com/ros-drivers/phidgets_drivers/blob/0a4e583e9d60c5c748e098837d11afafb8171ce5/phidgets_spatial/src/spatial_ros_i.cpp#L110

The error is then thrown here: 
https://github.com/ros2/rclcpp/blob/d0fa844a093b26e34dc0989e1b3ab94aee5ba896/rclcpp/include/rclcpp/parameter_value.hpp#L66

I propose moving the `declare_parameter` in the `try..catch` clause and additionally catch the `rclcpp::ParameterTypeException`.

We did that in https://github.com/angsa-robotics/phidgets_drivers

--- a/phidgets_magnetometer/src/magnetometer_ros_i.cpp
+++ b/phidgets_magnetometer/src/magnetometer_ros_i.cpp
@@ -89,19 +89,19 @@ MagnetometerRosI::MagnetometerRosI(const rclcpp::NodeOptions& options)
 
     // compass correction params (see
     // http://www.phidgets.com/docs/1044_User_Guide)
-    this->declare_parameter<double>("cc_mag_field");
-    this->declare_parameter<double>("cc_offset0");
-    this->declare_parameter<double>("cc_offset1");
-    this->declare_parameter<double>("cc_offset2");
-    this->declare_parameter<double>("cc_gain0");
-    this->declare_parameter<double>("cc_gain1");
-    this->declare_parameter<double>("cc_gain2");
-    this->declare_parameter<double>("cc_t0");
-    this->declare_parameter<double>("cc_t1");
-    this->declare_parameter<double>("cc_t2");
-    this->declare_parameter<double>("cc_t3");
-    this->declare_parameter<double>("cc_t4");
-    this->declare_parameter<double>("cc_t5");
+    this->declare_parameter("cc_mag_field", rclcpp::PARAMETER_DOUBLE);
+    this->declare_parameter("cc_offset0", rclcpp::PARAMETER_DOUBLE);
+    this->declare_parameter("cc_offset1", rclcpp::PARAMETER_DOUBLE);
+    this->declare_parameter("cc_offset2", rclcpp::PARAMETER_DOUBLE);
+    this->declare_parameter("cc_gain0", rclcpp::PARAMETER_DOUBLE);
+    this->declare_parameter("cc_gain1", rclcpp::PARAMETER_DOUBLE);
+    this->declare_parameter("cc_gain2", rclcpp::PARAMETER_DOUBLE);
+    this->declare_parameter("cc_t0", rclcpp::PARAMETER_DOUBLE);
+    this->declare_parameter("cc_t1", rclcpp::PARAMETER_DOUBLE);
+    this->declare_parameter("cc_t2", rclcpp::PARAMETER_DOUBLE);
+    this->declare_parameter("cc_t3", rclcpp::PARAMETER_DOUBLE);
+    this->declare_parameter("cc_t4", rclcpp::PARAMETER_DOUBLE);
+    this->declare_parameter("cc_t5", rclcpp::PARAMETER_DOUBLE);
 
     bool has_compass_params = false;
     double cc_mag_field = 0.0;
@@ -134,7 +134,7 @@ MagnetometerRosI::MagnetometerRosI(const rclcpp::NodeOptions& options)
         cc_T4 = this->get_parameter("cc_t4").get_value<double>();
         cc_T5 = this->get_parameter("cc_t5").get_value<double>();
         has_compass_params = true;
-    } catch (const rclcpp::exceptions::ParameterNotDeclaredException&)
+    } catch (const rclcpp::exceptions::ParameterUninitializedException &)
     {
     }
 

--- a/phidgets_spatial/src/spatial_ros_i.cpp
+++ b/phidgets_spatial/src/spatial_ros_i.cpp
@@ -107,19 +107,19 @@ SpatialRosI::SpatialRosI(const rclcpp::NodeOptions &options)
 
     // compass correction params (see
     // http://www.phidgets.com/docs/1044_User_Guide)
-    this->declare_parameter<double>("cc_mag_field");
-    this->declare_parameter<double>("cc_offset0");
-    this->declare_parameter<double>("cc_offset1");
-    this->declare_parameter<double>("cc_offset2");
-    this->declare_parameter<double>("cc_gain0");
-    this->declare_parameter<double>("cc_gain1");
-    this->declare_parameter<double>("cc_gain2");
-    this->declare_parameter<double>("cc_t0");
-    this->declare_parameter<double>("cc_t1");
-    this->declare_parameter<double>("cc_t2");
-    this->declare_parameter<double>("cc_t3");
-    this->declare_parameter<double>("cc_t4");
-    this->declare_parameter<double>("cc_t5");
+    this->declare_parameter("cc_mag_field", rclcpp::PARAMETER_DOUBLE);
+    this->declare_parameter("cc_offset0", rclcpp::PARAMETER_DOUBLE);
+    this->declare_parameter("cc_offset1", rclcpp::PARAMETER_DOUBLE);
+    this->declare_parameter("cc_offset2", rclcpp::PARAMETER_DOUBLE);
+    this->declare_parameter("cc_gain0", rclcpp::PARAMETER_DOUBLE);
+    this->declare_parameter("cc_gain1", rclcpp::PARAMETER_DOUBLE);
+    this->declare_parameter("cc_gain2", rclcpp::PARAMETER_DOUBLE);
+    this->declare_parameter("cc_t0", rclcpp::PARAMETER_DOUBLE);
+    this->declare_parameter("cc_t1", rclcpp::PARAMETER_DOUBLE);
+    this->declare_parameter("cc_t2", rclcpp::PARAMETER_DOUBLE);
+    this->declare_parameter("cc_t3", rclcpp::PARAMETER_DOUBLE);
+    this->declare_parameter("cc_t4", rclcpp::PARAMETER_DOUBLE);
+    this->declare_parameter("cc_t5", rclcpp::PARAMETER_DOUBLE);
 
     bool has_compass_params = false;
     double cc_mag_field = 0.0;
@@ -152,7 +152,7 @@ SpatialRosI::SpatialRosI(const rclcpp::NodeOptions &options)
         cc_T4 = this->get_parameter("cc_t4").get_value<double>();
         cc_T5 = this->get_parameter("cc_t5").get_value<double>();
         has_compass_params = true;
-    } catch (const rclcpp::exceptions::ParameterNotDeclaredException &)
+    } catch (const rclcpp::exceptions::ParameterUninitializedException &)
     {
     }
 

